Add switch to control `debug=true` on WCOSS2 for development testing

KateFriedman-NOAA opened the pull request:
# Description

This PR adds a switch (`DEBUG_POSTSCRIPT`) to control whether `debug=true` is set when submitting development rocoto jobs to PBS schedulers (currently just WCOSS2). This switch added to `config.base` will only work and be invoked on PBS platforms (e.g. WCOSS2). There isn't an equivalent flag to set for SLURM on the RDHPCS. Have added this new switch to documentation.

The `DEBUG_POSTSCRIPT` switch in `config.base` will be set to off/`"NO"` by default.

This PR does not touch the `debug=true` settings in the ecf scripts, which should remain for ops per NCO requests.

Resolves #619

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

Low res tests on Hera and WCOSS2.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

KateFriedman-NOAA committed:
Add DEBUG switch to config.base

Refs #619

KateFriedman-NOAA committed:
Add debug option to xml submissions

- When pbspro: add debug=true to native statement; similar
to what already happens but now users can turn it off
- When slurm: add SBATCH_DEBUG=2 (-vv, very verbose) to xml;
exports SBATCH_DEBUG to environment for all jobs

Refs #619

KateFriedman-NOAA committed:
Move SBATCH_DEBUG to xml envars

Set SBATCH_DEBUG via the envars in the xml instead
of as an ENTITY at the top. The setting now exports
with job submissions.

Refs #619

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/debug_switch

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/develop' into feature/debug_switch

* origin/develop:
  Return ocnanalrun npes resource setting back to previous value (#2386)
  Add new BUFR table file parm/product/bufr_ij9km.txt for GFSv17 C1152 (#2383)
  Add global-workflow infrastructure for ocean analysis recentering task (#2299)
  Clean out non-gfs top level variables (#2366)
  Forgotten templated  DO_VRFY_OCEANDA (#2379)
  Add Hercules support for the GSI monitor (#2373)
  Reformat snowDA templates to jinja2 (#2371)
  Reduce Jenkins messaging to GitHub (#2370)
  Adding debug option for all build scripts (#2326)
  Sort list of coupler restart files for restart time determination (#2360)
  Update stage IC to handle ocean perturbations (#2364)
  Turn on C48mx500_3DVarAOWCDA test on hera (#2363)
  Move Jenkinsfile into ci subdirectory (#2355)

KateFriedman-NOAA committed:
Remove SBATCH_DEBUG option from setup

Removing the debug switch setting for SLURM since
it doesn't work as expected.
Also added notes to DEBUG setting in configs to
note that it only works on WCOSS2.

Refs #619

aerorahul commented on parm/config/gefs/config.base:
```diff
@@ -75,6 +75,7 @@ export NMV="/bin/mv"
 export NLN="/bin/ln -sf"
 export VERBOSE="YES"
 export KEEPDATA="NO"
+export DEBUG="NO" # Only works on WCOSS2
```
`DEBUG_SCHEDULER` is perhaps better suited than a generic `DEBUG`. What do you think?

KateFriedman-NOAA replied:
Have changed to `DEBUG_SCHEDULER` per reviewer suggestion and offline discussion.

aerorahul reviewed (approved):
Looks good.
I would suggest using a targeted `DEBUG_xyz` to make it clear, this is debugging `xyz` and not a generic debug flag.

KateFriedman-NOAA committed:
Rename DEBUG variable and add to RTD

- Rename DEBUG to DEBUG_SCHEDULER.
- Add note that it's PBS only and that it sets debug=true.
- Add switch to documentation.

Refs #619

KateFriedman-NOAA committed:
Rename DEBUG switch to DEBUG_SCHEDULER

Rename variable in setup scripts

Refs #619

HenryRWinterbottom commented:
@KateFriedman-NOAA This looks good to me. 

Just a FYI, the following exists for SLURM: [OPT_DebugFlags](https://slurm.schedmd.com/slurm.conf.html#OPT_DebugFlags).

It is more convoluted than the above, but this is more intended as a FYI and beyond the scope of this PR (and likely anything else we might need).

KateFriedman-NOAA commented:
Thanks @HenryWinterbottom-NOAA ! I spoke with the Hera sysadmins about those options too but they are also meant for the admin side of the scheduler and not for users.

HenryRWinterbottom commented:
> Thanks @HenryWinterbottom-NOAA ! I spoke with the Hera sysadmins about those options too but they are also meant for the admin side of the scheduler and not for users.

@KateFriedman-NOAA That makes sense. The nerd in me has SLURM running on a small cluster here, so I forgot that it required admin permissions. Thank you for the heads up.

KateFriedman-NOAA committed:
Rename DEBUG_SCHEDULER to DEBUG_POSTSCRIPT

Based on reviewer feedback and discussion.

Refs #619

KateFriedman-NOAA committed:
Correct bug in .github docs.yaml

RTD sample link was not posted in updated PR.
Add missing curly brackets to line.

Refs #619

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/debug_switch
